File: android-app/baselibrary/src/main/java/com/musketeer/baselibrary/Activity/BaseFragment.java
/**   
* @Title: BaseFragment.java 
* @Package com.musketeer.lib.activity 
*
* @author musketeer zhongxuqi@163.com  
* @date 2014-11-29 下午4:39:56 
* @version V1.0   
*/
package com.musketeer.baselibrary.Activity;

import android.app.Fragment;
import android.content.Intent;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.Toast;

import com.musketeer.baselibrary.util.LogUtils;

import butterknife.ButterKnife;

/**
 * @author zhongxuqi
 *
 */
public abstract class BaseFragment extends Fragment implements OnClickListener, BaseFragmentUITask {
	public static final String TAG = "Musketeer_BaseFragment";

	protected View BaseView;
	
	/**屏幕的宽度*/
	protected int mScreenWidth;
	/**屏幕高度*/
	protected int mScreenHeight;
	/**屏幕密度*/
	protected float mDensity;
	
	private Toast mToast;

	@Override
	public void onCreate(Bundle savedInstanceState) {
		// TODO Auto-generated method stub
		super.onCreate(savedInstanceState);
		DisplayMetrics metric = new DisplayMetrics();
		getActivity().getWindowManager().getDefaultDisplay().getMetrics(metric);
		mScreenWidth = metric.widthPixels;
		mScreenHeight = metric.heightPixels;
		mDensity = metric.density;
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
		// TODO Auto-generated method stub
		super.onCreateView(inflater, container, savedInstanceState);
		setContentView(inflater, container, savedInstanceState);
		ButterKnife.bind(this, BaseView);
		initView();
		initEvent();
		initData();
		return BaseView;
	}
	
	protected void startActivity(Class<?> cls) {
		Intent intent=new Intent();
		intent.setClass(getActivity(), cls);
		startActivity(intent);
	}
	
	protected void startActivity(Class<?> cls,Bundle bundle) {
		Intent intent=new Intent();
		intent.putExtras(bundle);
		intent.setClass(getActivity(), cls);
		startActivity(intent);
	}
	
	protected void startActivityForResult(Class<?> cls,int requestCode) {
		Intent intent=new Intent();
		intent.setClass(getActivity(), cls);
		super.startActivityForResult(intent, requestCode);
	}
	
	protected void startActivityForResult(Class<?> cls,Bundle bundle,int requestCode) {
		Intent intent=new Intent();
		intent.putExtras(bundle);
		intent.setClass(getActivity(), cls);
		super.startActivityForResult(intent, requestCode);
	}
	
	protected void showCustomDebug(String text) {
		LogUtils.d(TAG, text);
	}
	
	protected void showCustomToast(String text) {
		if (mToast!=null) {
			mToast.cancel();
		}
		mToast=Toast.makeText(getActivity(), text, Toast.LENGTH_SHORT);
		mToast.show();
	}
	
	@Override
	public void onClick(View v) {
		// TODO Auto-generated method stub
		
	}
}

File: android-app/app/src/main/kotlin/com/muskeeter/base/fragment/BaseFragment.kt
package com.muskeeter.base.fragment

import android.app.Fragment
import android.content.Intent
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.muskeeter.base.acitivity.BaseActivity
import com.muskeeter.base.acitivity.BaseFragmentUITask
import com.muskeeter.base.acitivity.BaseUITask
import com.musketeer.scratchpaper.utils.LogUtils

/**
 * Created by zhongxuqi on 08/07/2017.
 */

abstract class BaseFragment: Fragment(), View.OnClickListener, BaseFragmentUITask {
    companion object {
        val TAG = "BaseFragment"
    }

    protected var BaseView: View? = null

    /**屏幕的宽度 */
    protected var mScreenWidth: Int = 0
    /**屏幕高度 */
    protected var mScreenHeight: Int = 0
    /**屏幕密度 */
    protected var mDensity: Float = 0.toFloat()

    private var mToast: Toast? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val metric = DisplayMetrics()
        activity.windowManager.defaultDisplay.getMetrics(metric)
        mScreenWidth = metric.widthPixels
        mScreenHeight = metric.heightPixels
        mDensity = metric.density
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        setContentView(inflater, container, savedInstanceState)
        initView()
        initEvent()
        initData()
        return BaseView
    }

    protected fun findViewById(id: Int): View? {
        return BaseView?.findViewById(id)
    }

    protected fun startActivity(cls: Class<*>) {
        val intent = Intent()
        intent.setClass(activity, cls)
        startActivity(intent)
    }

    protected fun startActivity(cls: Class<*>, bundle: Bundle) {
        val intent = Intent()
        intent.putExtras(bundle)
        intent.setClass(activity, cls)
        startActivity(intent)
    }

    protected fun startActivityForResult(cls: Class<*>, requestCode: Int) {
        val intent = Intent()
        intent.setClass(activity, cls)
        super.startActivityForResult(intent, requestCode)
    }

    protected fun startActivityForResult(cls: Class<*>, bundle: Bundle, requestCode: Int) {
        val intent = Intent()
        intent.putExtras(bundle)
        intent.setClass(activity, cls)
        super.startActivityForResult(intent, requestCode)
    }

    protected fun showCustomDebug(text: String) {
        LogUtils.d(BaseActivity.TAG, text)
    }

    protected fun showCustomToast(text: String) {
        if (mToast != null) {
            mToast!!.cancel()
        }
        mToast = Toast.makeText(activity, text, Toast.LENGTH_SHORT)
        mToast!!.show()
    }

    protected fun showCustomToast(resId: Int) {
        if (mToast != null) {
            mToast!!.cancel()
        }
        mToast = Toast.makeText(activity, resources.getString(resId), Toast.LENGTH_SHORT)
        mToast!!.show()
    }

    override fun onClick(v: View) {

    }
}
